src/data/real/liouville.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import data.real.irrational
+
+/-!
+# Liouville's theorem
+This file will contain a proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+
+At the moment, it contains the definition of a Liouville number and a proof that Liouville
+numbers are irrational.
+
+The commented imports below will appear as they are needed.  I (Damiano) leave them here for
+ease of reference.
+--import data.polynomial.denoms_clearable
+--import ring_theory.algebraic
+--import topology.algebra.polynomial
+--import analysis.calculus.mean_value
+-/
+
+section irrational
+
+/--
+A Liouville number `x` is a number such that for every natural number `n`, there exists `a, b ∈ ℤ`
+with `1 < b` such that `0 < |x - a/b| < 1/bⁿ`.  In the implementation, the condition `x ≠ a/b`
+replaces the traditional equivalent `0 < |x - a/b|`.
+-/
+def is_liouville (x : ℝ) := ∀ n : ℕ, ∃ a b : ℤ,
+  1 < b ∧ x ≠ a / b ∧ abs (x - a / b) < 1 / b ^ n
+
+lemma irrational_of_is_liouville {x : ℝ} (h : is_liouville x) : irrational x :=
+begin
+  rintros ⟨⟨a, b, bN0, cop⟩, rfl⟩,
+  change (is_liouville (a / b)) at h,
+  rcases h (b + 1) with ⟨p, q, q1, a0, a1⟩,
+  have qR0 : (0 : ℝ) < q := int.cast_pos.mpr (zero_lt_one.trans q1),
+  have b0 : (b : ℝ) ≠ 0 := ne_of_gt (nat.cast_pos.mpr bN0),
+  have bq0 : (0 : ℝ) < b * q := mul_pos (nat.cast_pos.mpr bN0) qR0,
+  rw [div_sub_div _ _ b0 (ne_of_gt qR0), abs_div, div_lt_div_iff (abs_pos.mpr (ne_of_gt bq0))
+    (pow_pos qR0 _), abs_of_pos bq0, one_mul] at a1,
+  rw [← int.cast_pow, ← int.cast_mul, ← int.cast_coe_nat, ← int.cast_mul, ← int.cast_mul,
+    ← int.cast_sub, ← int.cast_abs, ← int.cast_mul, int.cast_lt] at a1,
```

Comment:
You have two successive blocks of rewrites on `a1`. Why not convert them to one single big `rw`? 

Also, when reading that from outside vscode, it is very hard to follow a sequence of rewrites like that. To show the reader where you are going, you can just do your sequence of rewrites, but announce the result of the manipulations, with something like
```suggestion
replace a1 : abs (a * q - b * p) * q ^ (b + 1) < b * q, by
    rwa [div_sub_div _ _ b0 (ne_of_gt qR0), abs_div, div_lt_div_iff (abs_pos.mpr (ne_of_gt bq0))
      (pow_pos qR0 _), abs_of_pos bq0, one_mul,
      ← int.cast_pow, ← int.cast_mul, ← int.cast_coe_nat, ← int.cast_mul, ← int.cast_mul,
      ← int.cast_sub, ← int.cast_abs, ← int.cast_mul, int.cast_lt] at a1,
```